language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ServiceDeclarationNodeContext.java

```diff
@@ -134,7 +135,16 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
             Eg:
             (1) service / m<cursor>
             function ...
+            (2) service TestObj m<cursor>
+            (3) service /<cursor>
+            (4) service / <cursor>
              */
+            if (!node.onKeyword().isMissing() ||
+                    context.getNodeAtCursor().textRange().endOffset() == context.getCursorPositionInTree()
+                            && (context.getNodeAtCursor().toSourceCode().split(" ").length == 1
```

Comment:
Why do we need to check this?

Reply:
To distinguish between `service / o<cursor>` and `service /<cursor> `.  Here we should suggest `on` keyword for only for the first case.

Reply:
@TharushiJay